src/main/webapp/app/course/competencies/forms/common-course-competency-form.component.ts

```diff
@@ -144,85 +98,29 @@ export class CompetencyFormComponent implements OnInit, OnChanges {
         this.descriptionControl?.markAsDirty();
     }
 
-    ngOnChanges(): void {
-        this.initializeForm();
-        if (this.isEditMode && this.formData) {
-            this.setFormValues(this.formData);
-        }
-    }
-
     ngOnInit(): void {
-        this.initializeForm();
+        merge(this.titleControl!.valueChanges, this.descriptionControl!.valueChanges).subscribe(() => this.suggestTaxonomies());
     }
 
-    private initializeForm() {
-        if (this.form) {
-            return;
-        }
-        let initialTitle: string | undefined = undefined;
-        if (this.isEditMode && this.formData && this.formData.title) {
-            initialTitle = this.formData.title;
-        }
-        this.form = this.fb.nonNullable.group({
-            title: [undefined as string | undefined, [Validators.required, Validators.maxLength(255)], [titleUniqueValidator(this.competencyService, this.courseId, initialTitle)]],
-            description: [undefined as string | undefined, [Validators.maxLength(10000)]],
-            softDueDate: [undefined],
-            taxonomy: [undefined as CompetencyTaxonomy | undefined],
-            masteryThreshold: [DEFAULT_MASTERY_THRESHOLD, [Validators.min(0), Validators.max(100)]],
-            optional: [false],
-        });
-        this.selectedLectureUnitsInTable = [];
-
+    ngOnChanges(): void {
         merge(this.titleControl!.valueChanges, this.descriptionControl!.valueChanges).subscribe(() => this.suggestTaxonomies());
```

Comment:
Do we really have to subscribe to these value changes both onInit and onChanges? The tests crash locally for me unless I remove one of them.

I'd suggest removing the one in `ngOnChanges` since that gets called a lot -> lots of subscriptions